src/stream/DOT-Hip.cpp

```diff
@@ -0,0 +1,141 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-19, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "DOT.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#if defined(RAJA_ENABLE_HIP)
+
+#include "common/HipDataUtils.hpp"
+
+#include <iostream>
+
+
+namespace rajaperf
+{
+namespace stream
+{
+
+  //
+  // Define thread block size for HIP execution
+  //
+  const size_t block_size = 256;
+
+
+#define DOT_DATA_SETUP_HIP \
+  Real_ptr a; \
+  Real_ptr b; \
+\
+  allocAndInitHipDeviceData(a, m_a, iend); \
+  allocAndInitHipDeviceData(b, m_b, iend);
+
+#define DOT_DATA_TEARDOWN_HIP \
+  deallocHipDeviceData(a); \
+  deallocHipDeviceData(b);
+
+__global__ void dot(Real_ptr a, Real_ptr b,
+                    Real_ptr dprod, Real_type dprod_init,
+                    Index_type iend)
+{
+  HIP_DYNAMIC_SHARED( Real_type, pdot)
+
+  Index_type i = blockIdx.x * blockDim.x + threadIdx.x;
+
+  pdot[ threadIdx.x ] = dprod_init;
+  for ( ; i < iend ; i += gridDim.x * blockDim.x ) {
+    pdot[ threadIdx.x ] += a[ i ] * b[i];
+  }
+  __syncthreads();
+
+  for ( i = blockDim.x / 2; i > 0; i /= 2 ) {
+    if ( threadIdx.x < i ) {
+      pdot[ threadIdx.x ] += pdot[ threadIdx.x + i ];
+    }
+     __syncthreads();
+  }
+
+#if 1 // serialized access to shared data;
+  if ( threadIdx.x == 0 ) {
+    //atomicAdd(dprod, pdot[ 0 ] );
+    RAJA::atomic::atomicAdd(RAJA::atomic::hip_atomic{}, dprod, pdot[ 0 ] );
```

Comment:
Atomic namespace has been removed:

```suggestion
    RAJA::atomicAdd(RAJA::hip_atomic{}, dprod, pdot[ 0 ] );
```